rust/shalom_dart_codegen/src/lib.rs

```diff
@@ -63,6 +67,16 @@ mod ext_jinja_fns {
         }
     }
 
+    pub fn type_name_for_variable(variable: ViaDeserialize<VariableDefinition>) -> String {
+        let type_ref = match variable.0.ty {
+            FieldType::Named(type_ref) => type_ref,
+            FieldType::NonNullNamed(type_ref) => type_ref,
+            _ => unimplemented!("lists not implemented"),
+        };
+        let resolved = DEFAULT_SCALARS_MAP.get(&type_ref.name).unwrap();
```

Comment:
super duper wrong. you should resolve type type of the argument using schema context.

Reply:
ig this function should be provided with the schema ctx somehow **(not via deserialize)**